compiler/rustc_attr_parsing/src/attributes/stability.rs

```diff
@@ -471,3 +471,149 @@ pub(crate) fn parse_unstability<S: Stage>(
+            // same staged-api restriction as other stability attrs
+            reject_outside_std!(cx);
+
+            let ArgParser::List(list) = args else {
+                cx.expected_list(cx.attr_span);
+                return;
+            };
+
+            let mut feature_opt: Option<Symbol> = None;
+            let mut reason_opt: Option<Symbol> = None;
+            let mut since_opt: Option<StableSince> = None;
+
+            for param in list.mixed() {
+                let param_span = param.span();
+                let Some(param) = param.meta_item() else {
+                    cx.emit_err(session_diagnostics::UnsupportedLiteral {
+                        span: param_span,
+                        reason: UnsupportedLiteralReason::Generic,
+                        is_bytestr: false,
+                        start_point_span: cx.sess().source_map().start_point(param_span),
+                    });
+                    return;
+                };
+
+                let word = param.path().word();
+                match word.map(|i| i.name) {
+                    Some(sym::feature) => {
+                        if insert_value_into_option_or_error(
+                            cx,
+                            &param,
+                            &mut feature_opt,
+                            word.unwrap(),
+                        )
+                        .is_none()
+                        {
+                            return;
+                        }
+                    }
+                    Some(sym::reason) => {
+                        if insert_value_into_option_or_error(
+                            cx,
+                            &param,
+                            &mut reason_opt,
+                            word.unwrap(),
+                        )
+                        .is_none()
+                        {
+                            return;
+                        }
+                    }
+                    Some(sym::since) => {
+                        if insert_value_into_option_or_error(
+                            cx,
+                            &param,
+                            &mut None::<Symbol>,
+                            word.unwrap(),
+                        )
+                        .is_none()
+                        {
+                            return;
+                        }
+                        // parse version string
+                        if let Some(s) = param.args().name_value().and_then(|nv| nv.value_as_str())
+                        {
+                            if s.as_str() == VERSION_PLACEHOLDER {
+                                since_opt = Some(StableSince::Current);
+                            } else if let Some(v) = super::util::parse_version(s) {
+                                since_opt = Some(StableSince::Version(v));
+                            } else {
+                                let err = cx.emit_err(session_diagnostics::InvalidSince {
```

Comment:
maybe just error out in this case instead of putting a magic symbol in. If you do want to continue, return None but using sym::err doesn't seem like a great idea